java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -89,11 +94,28 @@ abstract class AbstractArcHost(
         initialParticles.toList().associateByTo(particleConstructors, { it.first }, { it.second })
     }
 
+    private val cacheMutex = Mutex()
+    private val runningMutex = Mutex()
```

Comment:
Might be worth moving these up to the variables that they guard?

Also, I think there's a guardedBy thing you can use? Not sure if that's helpful here.

Reply:
Done